packages/datadog-plugin-google-cloud-pubsub/src/pubsub-push-subscription.js

```diff
@@ -0,0 +1,110 @@
+    // Check for unwrapped Pub/Sub format (--push-no-wrapper-write-metadata)
+    if (req.headers['x-goog-pubsub-message-id']) {
+      log.debug('[PubSub] Detected unwrapped Pub/Sub push subscription')
+      this._createDeliverySpanAndActivate({ req, res })
+    } else {
+      log.warn(
+        '[PubSub] No x-goog-pubsub-* headers detected. pubsub.delivery spans will not be created. ' +
+        'Add --push-no-wrapper-write-metadata to your subscription.'
+      )
+    }
+  }
+
+  _createDeliverySpanAndActivate ({ req, res }) {
+    const messageData = this._parseMessage(req)
+    if (!messageData) return
+
+    const tracer = this.tracer || require('../../dd-trace')
+    if (!tracer || !tracer._tracer) return
+
+    const parentContext = tracer._tracer.extract('text_map', messageData.attrs) || undefined
+    const deliverySpan = this._createDeliverySpan(messageData, parentContext, tracer)
+    const finishDelivery = () => {
+      if (!deliverySpan.finished) {
+        deliverySpan.finish()
+      }
+    }
+
+    res.once('finish', finishDelivery)
+    res.once('close', finishDelivery)
+    res.once('error', (err) => {
+      deliverySpan.setTag('error', err)
+      finishDelivery()
+    })
+
+    const store = storage('legacy').getStore()
+    storage('legacy').enterWith({ ...store, span: deliverySpan, req, res })
+  }
+
+  _parseMessage (req) {
+    const subscription = req.headers['x-goog-pubsub-subscription-name']
+    const message = {
+      messageId: req.headers['x-goog-pubsub-message-id'],
+      publishTime: req.headers['x-goog-pubsub-publish-time']
+    }
+
+    const { projectId, topicName } = this._extractProjectTopic(req.headers, subscription)
+    return { message, subscription, attrs: req.headers, projectId, topicName }
+  }
+
+  _createDeliverySpan (messageData, parentContext, tracer) {
+    const { message, subscription, topicName } = messageData
+    const subscriptionName = subscription.split('/').pop() || subscription
+
+    const span = tracer._tracer.startSpan('pubsub.delivery', {
+      childOf: parentContext,
+      integrationName: 'google-cloud-pubsub',
+      tags: {
+        'span.kind': 'consumer',
+        component: 'google-cloud-pubsub',
+        '_dd.integration': 'google-cloud-pubsub',
+        'pubsub.method': 'delivery',
+        'pubsub.subscription': subscription,
+        'pubsub.message_id': message.messageId,
+        'pubsub.delivery_method': 'push',
+        'pubsub.topic': topicName,
+        service: this.config.service || `${tracer._tracer._service}-pubsub`,
+        '_dd.base_service': tracer._tracer._service,
+        '_dd.serviceoverride.type': 'integration'
+      }
+    })
+
+    span.setTag('resource.name', `Push Subscription ${subscriptionName}`)
+
+    return span
+  }
+
+  _extractProjectTopic (attrs, subscription) {
+    const topicName = attrs['pubsub.topic']
+    const projectId = subscription.match(/projects\/([^\\/]+)\/subscriptions/)
```

Comment:
Where does the `projectId` get used?

Notably the `String#match()` method either returns an array where the [1] index is the match, or a null if there is no match.

```js
"projects/abcde/subscriptions".match(/projects\/([^\\/]+)\/subscriptions/)
[
  'projects/abcde/subscriptions',
  'abcde',
  index: 0,
  input: 'projects/abcde/subscriptions',
  groups: undefined
]
```

With a name like projectId I'm assuming you would want to return the string at [1].